src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,549 @@
+        }
+
+        // add the async keyword
+        changes.insertModifierBefore(sourceFile, SyntaxKind.AsyncKeyword, functionToConvert);
+
+        function startTransformation(node: CallExpression, nodeToReplace: Node) {
+            const newNodes = transformExpression(node, transformer, node);
+            changes.replaceNodeWithNodes(sourceFile, nodeToReplace, newNodes);
+        }
+
+        for (const statement of returnStatements) {
+            if (isCallExpression(statement)) {
+                startTransformation(statement, statement);
+            }
+            else {
+                forEachChild(statement, function visit(node: Node) {
+                    if (isCallExpression(node)) {
+                        startTransformation(node, statement);
+                    }
+                    else if (!isFunctionLike(node)) {
+                        forEachChild(node, visit);
+                    }
+                });
+            }
+        }
+    }
+
+    // Returns the identifiers that are never reassigned in the refactor
+    function getConstIdentifiers(synthNamesMap: Map<SynthIdentifier>): Identifier[] {
+        const constIdentifiers: Identifier[] = [];
+        synthNamesMap.forEach((val) => {
+            if (val.numberOfAssignmentsOriginal === 0) {
+                constIdentifiers.push(val.identifier);
+            }
+        });
+        return constIdentifiers;
+    }
+
+
+    /*
+        Finds all of the expressions of promise type that should not be saved in a variable during the refactor
+    */
+    function getAllPromiseExpressionsToReturn(func: FunctionLikeDeclaration, checker: TypeChecker): Map<true> {
+        if (!func.body) {
+            return createMap<true>();
+        }
+
+        const setOfExpressionsToReturn: Map<true> = createMap<true>();
+
+        forEachChild(func.body, function visit(node: Node) {
+            if (isPromiseReturningExpression(node, checker, "then")) {
+                setOfExpressionsToReturn.set(getNodeId(node).toString(), true);
+                forEach((<CallExpression>node).arguments, visit);
+            }
+            else if (isPromiseReturningExpression(node, checker, "catch")) {
+                setOfExpressionsToReturn.set(getNodeId(node).toString(), true);
+                // if .catch() is the last call in the chain, move leftward in the chain until we hit something else that should be returned
+                forEachChild(node, visit);
+            }
+            else if (isPromiseReturningExpression(node, checker)) {
+                setOfExpressionsToReturn.set(getNodeId(node).toString(), true);
+                // don't recurse here, since we won't refactor any children or arguments of the expression
+            }
+            else {
+                forEachChild(node, visit);
+            }
+        });
+
+        return setOfExpressionsToReturn;
+    }
+
+    function isPromiseReturningExpression(node: Node, checker: TypeChecker, name?: string): boolean {

```

Comment:
For any of these predicate-type functions, an explanatory comment showing a 'true' and 'false' case would be really helpful. For this function, for example, it's not obvious to me what `name` is for